pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/bestpractices/UnusedPrivateMethodRule.java

```diff
@@ -107,4 +122,26 @@ private boolean privateAndNotExcluded(NameDeclaration mnd) {
                 && !node.hasImageEqualTo("writeObject") && !node.hasImageEqualTo("readResolve")
                 && !node.hasImageEqualTo("writeReplace");
     }
+
+    /**
+     * Checks whether the given node is annotated with annotation in the set.
+     * The node should be method name declaration.
+     *
+     * @param node
+     *            the node to check
+     * @return <code>true</code> if the annotation has been found
+     */
+    protected boolean hasNeglectAnnotation(MethodNameDeclaration node) {
```

Comment:
Maybe this should be enough of a common pattern (checking if a node has an annotation in a particular set) that we could move that logic to ASTMethodDeclaration someday. Otherwise it'll be duplicated in every rule that needs it...

Reply:
Same logic here.
https://github.com/pmd/pmd/blob/0967d5027539d8528deb025d6b6d0ecece41d8f9/pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/AbstractLombokAwareRule.java#L89

Reply:
I added #956 to track this.